packages/api/src/command/medical/patient/handle-data-contributions.ts

```diff
@@ -51,26 +52,38 @@ export async function handleDataContribution({
     fhirBundle: validatedBundle,
     destinationKey: fhirBundleDestinationKey,
   });
-  const patientDataPromise = async () => {
-    return createOrUpdateConsolidatedPatientData({
-      cxId,
-      patientId: patient.id,
-      fhirBundle: validatedBundle,
-    });
-  };
+  const consolidatedDataUploadResults = await createOrUpdateConsolidatedPatientData({
+    cxId,
+    patientId: patient.id,
+    fhirBundle: validatedBundle,
+  });
+
   const convertAndUploadCdaPromise = async () => {
     const isValidForCdaConversion = hasCompositionResource(validatedBundle);
     if (isValidForCdaConversion) {
-      await convertFhirToCda({
+      const converted = await convertFhirToCda({
+        cxId,
+        validatedBundle,
+      });
+      await uploadCdaDocuments({
         cxId,
         patientId,
+        cdaBundles: converted,
+        organization: fhirOrganization,
         docId,
-        validatedBundle,
       });
     }
   };
+  const createAndUploadCcdPromise = async () => {
+    // TODO: To minimize generating CCDs, make it a delayed job (run it ~5min after it was initiated, only once for all requests within that time window)
+    await processCcdRequest(patient, fhirOrganization);
+  };
 
-  return Promise.all([patientDataPromise(), convertAndUploadCdaPromise()]);
+  return Promise.all([
+    consolidatedDataUploadResults,
+    createAndUploadCcdPromise(),
+    convertAndUploadCdaPromise(),
+  ]);
```

Comment:
No concern if one of these fail but other(s) finish executing successfully, right?

Reply:
Also, this is returning an array of stuff, but:
1. The only place that calls it only uses the first item;
2. If we want to return multiple things, its better to return an object so its clear which data is what on the caller side, instead of relying on the order of the items in the array.

Reply:
fixed